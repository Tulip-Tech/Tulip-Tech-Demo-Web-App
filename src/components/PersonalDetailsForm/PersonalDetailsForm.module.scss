@use 'src/styles/variables';
@use 'src/styles/theme';

.title {
  margin-bottom: 24px;
  font-weight: var(--body-font-weight-bold);
  font-size: 24px;
}

.continue {
  margin: variables.$base-spacing 0;
}
