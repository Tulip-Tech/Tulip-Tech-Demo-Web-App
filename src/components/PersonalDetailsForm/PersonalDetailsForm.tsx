import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './PersonalDetailsForm.module.scss';

import TextField from '#components/TextField/TextField';
import Button from '#components/Button/Button';
import Dropdown from '#components/Dropdown/Dropdown';
import Checkbox from '#components/Checkbox/Checkbox';
import Radio from '#components/Radio/Radio';
import DateField from '#components/DateField/DateField';
import LoadingOverlay from '#components/LoadingOverlay/LoadingOverlay';
import FormFeedback from '#components/FormFeedback/FormFeedback';
import { testId } from '#src/utils/common';
import type { FormErrors } from '#types/form';
import type { PersonalDetailsFormData, CleengCaptureField, CleengCaptureQuestionField } from '#types/account';

type Props = {
  onSubmit: React.FormEventHandler<HTMLFormElement>;
  onChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  setValue: (key: keyof PersonalDetailsFormData, value: string) => void;
  error?: string;
  errors: FormErrors<PersonalDetailsFormData>;
  values: PersonalDetailsFormData;
  submitting: boolean;
  fields: Record<string, CleengCaptureField>;
  questions: CleengCaptureQuestionField[];
  onQuestionChange: React.ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>;
  questionValues: Record<string, string>;
  questionErrors: Record<string, string>;
};

const PersonalDetailsForm: React.FC<Props> = ({
  onSubmit,
  onChange,
  setValue,
  values,
  errors,
  submitting,
  fields,
  questions,
  onQuestionChange,
  questionValues,
  questionErrors,
}: Props) => {
  const { t } = useTranslation('account');
  const renderQuestion = ({ value, key, question, required }: CleengCaptureQuestionField) => {
    const values = value?.split(';') || [];
    const props = {
      name: key,
      onChange: onQuestionChange,
      error: !!questionErrors[key],
      helperText: questionErrors[key],
      required,
      key,
    };

    if (values.length === 1) {
      return <Checkbox checked={!!questionValues[key]} value={values[0]} header={question} label={values[0]} {...props} />;
    } else if (values.length === 2) {
      return <Radio values={values} value={questionValues[key]} header={question} {...props} />;
    } else if (values.length > 2) {
      return <Dropdown options={values} value={questionValues[key]} label={question} defaultLabel={t('personal_details.select_answer')} {...props} fullWidth />;
    }

    return null;
  };

  return (
    <form onSubmit={onSubmit} data-testid={testId('personal_details-form')} noValidate>
      <h2 className={styles.title}>{t('personal_details.title')}</h2>
      {errors.form ? <FormFeedback variant="error">{errors.form}</FormFeedback> : null}
      {fields.firstNameLastName?.enabled ? (
        <React.Fragment>
          <TextField
            value={values.firstName}
            onChange={onChange}
            label={t('personal_details.fist_name')}
            placeholder={t('personal_details.fist_name')}
            error={!!errors.firstName || !!errors.form}
            helperText={errors.firstName}
            required={fields.firstNameLastName.required}
            name="firstName"
          />
          <TextField
            value={values.lastName}
            onChange={onChange}
            label={t('personal_details.last_name')}
            placeholder={t('personal_details.last_name')}
            error={!!errors.lastName || !!errors.form}
            helperText={errors.lastName}
            required={fields.firstNameLastName.required}
            name="lastName"
          />
        </React.Fragment>
      ) : null}
      {fields.companyName?.enabled ? (
        <TextField
          value={values.companyName}
          onChange={onChange}
          label={t('personal_details.company_name')}
          placeholder={t('personal_details.company_name')}
          error={!!errors.companyName || !!errors.form}
          helperText={errors.companyName}
          required={fields.companyName.required}
          name="companyName"
        />
      ) : null}
      {fields.address?.enabled ? (
        <React.Fragment>
          <TextField
            value={values.address}
            onChange={onChange}
            label={t('personal_details.address')}
            placeholder={t('personal_details.address')}
            error={!!errors.address || !!errors.form}
            helperText={errors.address}
            required={fields.address.required}
            name="address"
          />
          <TextField
            value={values.address2}
            onChange={onChange}
            label={t('personal_details.address2')}
            placeholder={t('personal_details.address2')}
            error={!!errors.address2 || !!errors.form}
            helperText={errors.address2}
            name="address2"
          />
          <TextField
            value={values.city}
            onChange={onChange}
            label={t('personal_details.city')}
            placeholder={t('personal_details.city')}
            error={!!errors.city || !!errors.form}
            helperText={errors.city}
            required={fields.address.required}
            name="city"
          />
          <TextField
            value={values.state}
            onChange={onChange}
            label={t('personal_details.state')}
            placeholder={t('personal_details.state')}
            error={!!errors.state || !!errors.form}
            helperText={errors.state}
            required={fields.address.required}
            name="state"
          />
          <TextField
            value={values.postCode}
            onChange={onChange}
            label={t('personal_details.post_code')}
            placeholder={t('personal_details.post_code')}
            error={!!errors.postCode || !!errors.form}
            helperText={errors.postCode}
            required={fields.address.required}
            name="postCode"
          />
        </React.Fragment>
      ) : null}
      {fields.phoneNumber?.enabled ? (
        <TextField
          value={values.phoneNumber}
          onChange={onChange}
          label={t('personal_details.phone_number')}
          placeholder={t('personal_details.phone_number')}
          error={!!errors.phoneNumber || !!errors.form}
          helperText={errors.phoneNumber}
          required={fields.phoneNumber.required}
          name="phoneNumber"
        />
      ) : null}
      {fields.birthDate?.enabled ? (
        <DateField
          value={values.birthDate}
          onChange={(value) => setValue('birthDate', value)}
          label={t('personal_details.birth_date')}
          placeholder={t('personal_details.birth_date')}
          error={!!errors.birthDate || !!errors.form}
          helperText={errors.birthDate}
          required={fields.birthDate.required}
          name="birthDate"
        />
      ) : null}
      {questions.map((question) => renderQuestion(question))}
      <Button
        className={styles.continue}
        type="submit"
        label={t('personal_details.continue')}
        variant="contained"
        color="primary"
        size="large"
        disabled={submitting}
        fullWidth
      />
      {submitting && <LoadingOverlay transparentBackground inline />}
    </form>
  );
};

export default PersonalDetailsForm;
