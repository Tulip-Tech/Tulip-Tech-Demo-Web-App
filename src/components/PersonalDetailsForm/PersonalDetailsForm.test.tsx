import { render } from '@testing-library/react';

import PersonalDetailsForm from '#components/PersonalDetailsForm/PersonalDetailsForm';

const noop = () => {
  /**/
};
const values = {
  firstName: 'Vince',
  lastName: 'Guy',
  birthDate: '27-Feb-1985',
  companyName: 'JW Player',
  phoneNumber: '+1 555 555 5555',
  address: '123 Nowhere St.',
  address2: 'Apt 1W',
  city: 'Chicago',
  state: 'IL',
  postCode: '60601',
  country: 'United States',
};
const fields = {
  firstNameLastName: {
    key: 'fnln',
    enabled: true,
    required: true,
    answer: '',
  },
  companyName: {
    key: 'c',
    enabled: true,
    required: false,
    answer: '',
  },
  address: {
    key: 'a',
    enabled: true,
    required: true,
    answer: '',
  },
  phoneNumber: {
    key: 'pn',
    enabled: true,
    required: true,
    answer: '',
  },
  birthDate: {
    key: 'dob',
    enabled: true,
    required: true,
    answer: '',
  },
};
const questions = [
  {
    key: 'key1',
    question: 'Which number is 2?',
    value: '1;2;3',
    answer: '',
    required: true,
    enabled: true,
  },
  {
    key: 'key2',
    question: 'Is this a question?',
    value: 'Yes;No',
    answer: '',
    required: false,
    enabled: true,
  },
  {
    key: 'ccc',
    question: 'Check this off.',
    value: 'CheckMe',
    answer: '',
    required: true,
    enabled: true,
  },
];
const questionValues = {
  key1: '3',
  key2: 'No',
  ccc: 'true',
};

describe('<PersonalDetailsForm>', () => {
  test('Renders without crashing', () => {
    const { container } = render(
      <PersonalDetailsForm
        onSubmit={noop}
        onChange={noop}
        setValue={noop}
        errors={{}}
        values={values}
        submitting={false}
        fields={fields}
        questions={questions}
        onQuestionChange={noop}
        questionValues={questionValues}
        questionErrors={{}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test('Renders with errors', () => {
    const { container } = render(
      <PersonalDetailsForm
        onSubmit={noop}
        onChange={noop}
        setValue={noop}
        errors={{
          firstName: 'this is wrong',
          lastName: 'also this',
          address: 'Do you know where you live',
          address2: 'Come on',
          city: 'stop it',
          companyName: 'So much errors',
          country: 'usa is an error',
          birthDate: 'February 30th? Really?',
          phoneNumber: 'Invalid',
          state: 'Confusion',
          postCode: 'Post Code should be 5 centimeters',
          form: 'This is a form error',
        }}
        values={values}
        submitting={false}
        fields={fields}
        questions={questions}
        onQuestionChange={noop}
        questionValues={questionValues}
        questionErrors={{
          key1: 'question 1 is invalid',
          key2: 'Invalid answer',
          ccc: 'No way Jose',
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
