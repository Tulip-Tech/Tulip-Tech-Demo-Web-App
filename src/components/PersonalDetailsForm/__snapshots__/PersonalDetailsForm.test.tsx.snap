// Vitest Snapshot v1

exports[`<PersonalDetailsForm> > Renders with errors 1`] = `
<div>
  <form
    data-testid="personal_details-form"
    novalidate=""
  >
    <h2
      class="_title_8a9484"
    >
      personal_details.title
    </h2>
    <div
      class="_formFeedback_99f247 _error_99f247"
    >
      This is a form error
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_firstName"
      >
        personal_details.fist_name
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_firstName"
          name="firstName"
          placeholder="personal_details.fist_name"
          required=""
          type="text"
          value="Vince"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        this is wrong
      </div>
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_lastName"
      >
        personal_details.last_name
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_lastName"
          name="lastName"
          placeholder="personal_details.last_name"
          required=""
          type="text"
          value="Guy"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        also this
      </div>
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_companyName"
      >
        personal_details.company_name
        <span>
          optional
        </span>
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_companyName"
          name="companyName"
          placeholder="personal_details.company_name"
          type="text"
          value="JW Player"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        So much errors
      </div>
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_address"
      >
        personal_details.address
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_address"
          name="address"
          placeholder="personal_details.address"
          required=""
          type="text"
          value="123 Nowhere St."
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        Do you know where you live
      </div>
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_address2"
      >
        personal_details.address2
        <span>
          optional
        </span>
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_address2"
          name="address2"
          placeholder="personal_details.address2"
          type="text"
          value="Apt 1W"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        Come on
      </div>
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_city"
      >
        personal_details.city
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_city"
          name="city"
          placeholder="personal_details.city"
          required=""
          type="text"
          value="Chicago"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        stop it
      </div>
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_state"
      >
        personal_details.state
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_state"
          name="state"
          placeholder="personal_details.state"
          required=""
          type="text"
          value="IL"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        Confusion
      </div>
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_postCode"
      >
        personal_details.post_code
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_postCode"
          name="postCode"
          placeholder="personal_details.post_code"
          required=""
          type="text"
          value="60601"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        Post Code should be 5 centimeters
      </div>
    </div>
    <div
      class="_textField_e16c1b _error_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_phoneNumber"
      >
        personal_details.phone_number
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_phoneNumber"
          name="phoneNumber"
          placeholder="personal_details.phone_number"
          required=""
          type="text"
          value="+1 555 555 5555"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        Invalid
      </div>
    </div>
    <div
      class="_dateField_eb6974 _error_eb6974"
    >
      <label
        class="_label_eb6974"
        for="text-field_1235_birthDate"
      >
        personal_details.birth_date
      </label>
      <div
        class="_container_eb6974"
      >
        <input
          class="_input_eb6974"
          id="text-field_1235_birthDate"
          maxlength="2"
          name="date"
          placeholder="dd"
          type="number"
          value="27"
        />
         / 
        <input
          class="_input_eb6974"
          id="text-field_1235_birthDate"
          maxlength="2"
          name="month"
          placeholder="mm"
          type="number"
          value="2"
        />
         / 
        <input
          class="_input_eb6974"
          id="text-field_1235_birthDate"
          maxlength="4"
          name="year"
          placeholder="yyyy"
          type="number"
          value="1985"
        />
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        February 30th? Really?
      </div>
    </div>
    <div
      class="_container_bb73c6 _fullWidth_bb73c6 _error_bb73c6 _medium_bb73c6"
    >
      <label
        class="_label_bb73c6"
        for="1235"
      >
        Which number is 2?
      </label>
      <div
        class="_dropdown_bb73c6 _fullWidth_bb73c6"
      >
        <select
          aria-required="true"
          class="_select_bb73c6"
          id="1235"
          name="key1"
        >
          <option
            class="_option_bb73c6"
            disabled=""
            value=""
          >
            personal_details.select_answer
          </option>
          <option
            class="_option_bb73c6"
            value="1"
          >
            1
          </option>
          <option
            class="_option_bb73c6"
            value="2"
          >
            2
          </option>
          <option
            class="_option_bb73c6"
            value="3"
          >
            3
          </option>
        </select>
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        question 1 is invalid
      </div>
    </div>
    <div>
      <div
        class="_header_d50e6e"
      >
        Is this a question?
        <span>
          optional
        </span>
      </div>
      <div
        class="_radio_d50e6e"
      >
        <input
          id="radio_1235_key20"
          name="key2"
          type="radio"
          value="Yes"
        />
        <label
          for="radio_1235_key20"
        >
          Yes
        </label>
      </div>
      <div
        class="_radio_d50e6e"
      >
        <input
          checked=""
          id="radio_1235_key21"
          name="key2"
          type="radio"
          value="No"
        />
        <label
          for="radio_1235_key21"
        >
          No
        </label>
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0"
      >
        Invalid answer
      </div>
    </div>
    <div
      class="_checkbox_531f07 _error_531f07"
    >
      <div
        class="_header_531f07"
      >
        Check this off.
      </div>
      <div
        class="_row_531f07"
      >
        <input
          aria-required="true"
          checked=""
          id="check-box_1235_ccc"
          name="ccc"
          type="checkbox"
          value="CheckMe"
        />
        <label
          for="check-box_1235_ccc"
        >
          * 
          CheckMe
        </label>
      </div>
      <div
        class="_helperText_da58e0 _error_da58e0 _helperTextError_531f07"
      >
        No way Jose
      </div>
    </div>
    <button
      aria-disabled="false"
      class="_button_f8f296 _continue_8a9484 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
      type="submit"
    >
      <span>
        personal_details.continue
      </span>
    </button>
  </form>
</div>
`;

exports[`<PersonalDetailsForm> > Renders without crashing 1`] = `
<div>
  <form
    data-testid="personal_details-form"
    novalidate=""
  >
    <h2
      class="_title_8a9484"
    >
      personal_details.title
    </h2>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_firstName"
      >
        personal_details.fist_name
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_firstName"
          name="firstName"
          placeholder="personal_details.fist_name"
          required=""
          type="text"
          value="Vince"
        />
      </div>
    </div>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_lastName"
      >
        personal_details.last_name
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_lastName"
          name="lastName"
          placeholder="personal_details.last_name"
          required=""
          type="text"
          value="Guy"
        />
      </div>
    </div>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_companyName"
      >
        personal_details.company_name
        <span>
          optional
        </span>
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_companyName"
          name="companyName"
          placeholder="personal_details.company_name"
          type="text"
          value="JW Player"
        />
      </div>
    </div>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_address"
      >
        personal_details.address
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_address"
          name="address"
          placeholder="personal_details.address"
          required=""
          type="text"
          value="123 Nowhere St."
        />
      </div>
    </div>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_address2"
      >
        personal_details.address2
        <span>
          optional
        </span>
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_address2"
          name="address2"
          placeholder="personal_details.address2"
          type="text"
          value="Apt 1W"
        />
      </div>
    </div>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_city"
      >
        personal_details.city
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_city"
          name="city"
          placeholder="personal_details.city"
          required=""
          type="text"
          value="Chicago"
        />
      </div>
    </div>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_state"
      >
        personal_details.state
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_state"
          name="state"
          placeholder="personal_details.state"
          required=""
          type="text"
          value="IL"
        />
      </div>
    </div>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_postCode"
      >
        personal_details.post_code
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_postCode"
          name="postCode"
          placeholder="personal_details.post_code"
          required=""
          type="text"
          value="60601"
        />
      </div>
    </div>
    <div
      class="_textField_e16c1b"
    >
      <label
        class="_label_e16c1b"
        for="text-field_1235_phoneNumber"
      >
        personal_details.phone_number
      </label>
      <div
        class="_container_e16c1b"
      >
        <input
          class="_input_e16c1b"
          id="text-field_1235_phoneNumber"
          name="phoneNumber"
          placeholder="personal_details.phone_number"
          required=""
          type="text"
          value="+1 555 555 5555"
        />
      </div>
    </div>
    <div
      class="_dateField_eb6974"
    >
      <label
        class="_label_eb6974"
        for="text-field_1235_birthDate"
      >
        personal_details.birth_date
      </label>
      <div
        class="_container_eb6974"
      >
        <input
          class="_input_eb6974"
          id="text-field_1235_birthDate"
          maxlength="2"
          name="date"
          placeholder="dd"
          type="number"
          value="27"
        />
         / 
        <input
          class="_input_eb6974"
          id="text-field_1235_birthDate"
          maxlength="2"
          name="month"
          placeholder="mm"
          type="number"
          value="2"
        />
         / 
        <input
          class="_input_eb6974"
          id="text-field_1235_birthDate"
          maxlength="4"
          name="year"
          placeholder="yyyy"
          type="number"
          value="1985"
        />
      </div>
    </div>
    <div
      class="_container_bb73c6 _fullWidth_bb73c6 _medium_bb73c6"
    >
      <label
        class="_label_bb73c6"
        for="1235"
      >
        Which number is 2?
      </label>
      <div
        class="_dropdown_bb73c6 _fullWidth_bb73c6"
      >
        <select
          aria-required="true"
          class="_select_bb73c6"
          id="1235"
          name="key1"
        >
          <option
            class="_option_bb73c6"
            disabled=""
            value=""
          >
            personal_details.select_answer
          </option>
          <option
            class="_option_bb73c6"
            value="1"
          >
            1
          </option>
          <option
            class="_option_bb73c6"
            value="2"
          >
            2
          </option>
          <option
            class="_option_bb73c6"
            value="3"
          >
            3
          </option>
        </select>
      </div>
    </div>
    <div>
      <div
        class="_header_d50e6e"
      >
        Is this a question?
        <span>
          optional
        </span>
      </div>
      <div
        class="_radio_d50e6e"
      >
        <input
          id="radio_1235_key20"
          name="key2"
          type="radio"
          value="Yes"
        />
        <label
          for="radio_1235_key20"
        >
          Yes
        </label>
      </div>
      <div
        class="_radio_d50e6e"
      >
        <input
          checked=""
          id="radio_1235_key21"
          name="key2"
          type="radio"
          value="No"
        />
        <label
          for="radio_1235_key21"
        >
          No
        </label>
      </div>
    </div>
    <div
      class="_checkbox_531f07"
    >
      <div
        class="_header_531f07"
      >
        Check this off.
      </div>
      <div
        class="_row_531f07"
      >
        <input
          aria-required="true"
          checked=""
          id="check-box_1235_ccc"
          name="ccc"
          type="checkbox"
          value="CheckMe"
        />
        <label
          for="check-box_1235_ccc"
        >
          * 
          CheckMe
        </label>
      </div>
    </div>
    <button
      aria-disabled="false"
      class="_button_f8f296 _continue_8a9484 _primary_f8f296 _fullWidth_f8f296 _large_f8f296"
      type="submit"
    >
      <span>
        personal_details.continue
      </span>
    </button>
  </form>
</div>
`;
